components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -323,6 +322,36 @@ func generateStatefulSet(instance *v1beta1.Notebook) *appsv1.StatefulSet {
 			},
 		}
 	}
+	if len(container.Ports) == 1 {
```

Comment:
Here you assume there is only one container running in the pod, while in fact there are two; istio-proxy and notebook container. As result, the liveness- and readiness probe will only be added to the istio-proxy container and the notebook container will stay untouched.

Reply:
istio-proxy is added by istio sidecar injector.  And only when pod is created, the istio sidecar injector can add new containers. 
The controller watches the change of notebook resource and creates the correlated statefulset. The controller can only find one container when it generate statefulset definitions. FYI

Reply:
We could also target the port based off the name `notebook-port`

Reply:
Oops, sorry I did't know that :blush:. Thanks for the information